### ISSUE TITLE
database: database_maxage might remove all stored values from item

### ISSUE THREAD

**Issue by bmxp** [smarthomeNG/plugins#598]
Consider the following item:

```
Solltemperatur:
    type: num
    cache: True
```

The database plugin should drop in here to remove ``cache: True`` attribute thus the item is changed to:

```
Solltemperatur:
    type: num
    database: init
    database_maxage: 7
```

In case there is not much of change - let's assume no change for the last 14 days, then the attribute ``database_maxage: 7`` will ensure that all data for this item is removed after 7 days. 
But when restarting SmartHomeNG after 7 days, this will lead to a missing init value for the item since all database entries are already deleted.

A fix should be provided that ensures that at least the last data entry to the database is kept to provide an init value for a later restart of SmartHomeNG

**Comment by bmxp** [comment]
This should be fixed as of release 1.9.3. 

Please test and report if there are conditions when this leads to problems

**Comment by msinn** [comment]
@bmxp Was this fixed in v1.9.3?

**Comment by aschwith** [comment]
Tested the implementations made in the database plugin (tested with V1.10 although the adaptations were made longtime before). Issue is fixed. Ticket can be closed.
